### PR TITLE
Changes to unfccc client

### PR DESCRIPTION
**Pull request by rossf7**

### CODE REVIEW

File: unfccc/client.go [hunk 1]
```diff
@@ -3,64 +3,43 @@ package unfccc
 import (
 	"context"
 	"encoding/json"
-	"fmt"
-	"os"
+	"io/ioutil"
```
Comment (rossf7) [inline unfccc/client.go:6]: As of go 1.16 [os.ReadFile is preferred](https://pkg.go.dev/io/ioutil) but it causes CI to fail :(

ioutil will be around for a while so I switched to that.

File: unfccc/client.go [hunk 2]
```diff
@@ -3,64 +3,43 @@ package unfccc
...
 	var regionJSON []byte
-
-	a := &ApiClient{}
-
-	// figure out how to load JSON file
-	regionJSON, err := os.ReadFile(pathtoJSONFile)
+	regionJSON, err := ioutil.ReadFile(pathtoJSONFile)
 	if err != nil {
-		panic(err)
+		return nil, err
```
Comment (rossf7) [inline unfccc/client.go:22]: Not sure what I was thinking with the panics :)

File: unfccc/errors.go [hunk 3]
```diff
@@ -5,5 +5,5 @@ import (
 )
 
 var (
-	ErrNoMatchingRegion error = errors.New("No region matched that region code")
+	ErrNoMatchingRegion error = errors.New("no region matched that region code")
```
Comment (rossf7) [inline unfccc/errors.go:8]: Go is opinionated about errors and they should be lower case. Fixes a warning from golint.

File: examples/unfccc/main.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+func main() {
+	c, err := unfccc.New("../../unfccc/sampleRegionData.json")
+	if err != nil {
+		log.Fatalln("could not make provider", err)
+	}
+	a, err := c.GetCarbonIntensity(context.Background(), "de")
+	log.Println(a, err)
+}
```
Comment (rossf7) [inline examples/unfccc/main.go:17]: I added the example to align with the other providers.

**Comment by rossf7** [comment]
Hey @mrchrisadams hope you don't mind but this proposes some changes for the new provider.

File: examples/unfccc/main.go (hunk 4 above)
Comment (mrchrisadams) [reply examples/unfccc/main.go:17]: Awesome, thanks Ross :)

**Review by mrchrisadams — APPROVED** [review]
Thanks for this Ross - I'm happy to merge this in.

**Pull request merged**